### ISSUE TITLE
Error from runc: `cannot set limits on the memory cgroup, as the container has not joined it`

### ISSUE THREAD

**Issue by jvshahid** [cloudfoundry/garden-runc-release#74]
# Description

Error from runc: `cannot set limits on the memory cgroup, as the container has not joined it`

# Environment

* garden-runc-release version: `v1.12.1`
* IaaS: `AWS`
* Stemcell version: `ubuntu-trusty: 3541.12`
* Kernel version: `4.4.0-1049-aws`

# Steps to reproduce

We don't have repro steps yet. The error happened after a deploy:

- cell rep started up at 1524090639 = 2018-04-18T22:30:39Z, after passing its garden healthcheck
- cell rep took on this staging task two seconds later, at 1524090641 = 2018-04-18T22:30:41Z
- container creation then failed with:
```
{"timestamp":"1524090641.830159187","source":"rep","message":"rep.executing-container-operation.task-processor.run-container.containerstore-create.node-create.failed-to-create-container-in-garden","log_level":2,"data":{"container-guid":"35f7ad82-9831-4e09-b7e3-9c41df9dae22","container-state":"reserved","error":"runc run: exit status 1: container_linux.go:348: starting container process caused \"process_linux.go:402: container init caused \\\"process_linux.go:367: setting cgroup config for procHooks process caused \\\\\\\"cannot set limits on the memory cgroup, as the container has not joined it\\\\\\\"\\\"\"\n","guid":"35f7ad82-9831-4e09-b7e3-9c41df9dae22","session":"20.1.3.2.1"}}
```

# Logs

Logs and more context can be found in [this tracker story](https://www.pivotaltracker.com/story/show/156914550). **Note** output of `garden-ordance-survey` is also included in the story comments.

# Cause

*Suspicion* : it appears that the garden memory cgroup wasn't available at container creation time (Tom G).

# Resolution

Subsequent container creations succeeded as soon as 3 seconds after this failure.

Note from @BooleanCat : It might be worth trying to repro by slamming garden with container creations while it's starting up.

**Comment by Callisto13** [comment]
This issue was caused by interference from another job on the VM.

tldr: the FIM plugin was unmouting all cgroups on start, so when runc looked at the mount table it could not find the subsystem on which to set limits.

For full details see the tracker story.

**Comment by Callisto13** [comment]
The team responsible for the ClamAV and FIM projects are working to ensure that their cgroup setup is handled more delicately.
Closing as there is no change required in Garden-Runc-Release.
